src/Api/Order.php

```diff
@@ -197,17 +206,31 @@ public static function check_post_address( $params ) {
 	 */
 	public static function fields_have_errors( $params ) {
 		$errors = 0;
+
 		$required_billing = count( self::$billing_required_fields );
 		$required_shipping = count( self::$shipping_required_fields );
 
 		$array_billing = array_flip( self::$billing_required_fields );
 		$array_shipping = array_flip( self::$shipping_required_fields );
 
-		if ( count( array_intersect_key( $params[ self::BILLING_KEY ], $array_billing ) ) < $required_billing ) {
+		$array_billing_match = [];
+		$array_shipping_match = [];
+
+		// We need to pre-process the parameter arrays, due to a difference between the field and the final database key.
+		foreach ( $params[ self::BILLING_KEY ] as $key => $value ) {
```

Comment:
Can we refactor this into separate method like: 

```php
public static function fields_have_errors( $params ) { 
   $error += count_errors( self::BILLING_KEY,  self::$billing_required_fields);
}

protected function count_errors( $keys, required ){
$flip = array_ship( $required );
$tmp = [];
foreach ( $keys as $key => $value ) {
   $tmp[ $keys . '_' . $key ] = $value;
}
return (int) ( count( array_intersect_key( $tmp, $flip ) ) < count( $required) );
```

That will decrease the number of repeated lines on this method.